chain/types/ethtypes/eth_transactions.go

```diff
@@ -79,10 +84,18 @@ func EthTxFromSignedEthMessage(smsg *types.SignedMessage) (EthTx, error) {
 		return EthTx{}, xerrors.Errorf("failed to recover signature: %w", err)
 	}
 
+	from, err := EthAddressFromFilecoinAddress(smsg.Message.From)
```

Comment:
That check above lets us convert the address here without having to potentially look it up in the state.